app/components/project-preferences/provider/index.ts

```diff
@@ -0,0 +1,214 @@
+import Component from '@glimmer/component';
+import { inject as service } from '@ember/service';
+import ENUMS from 'irene/enums';
+import ENV from 'irene/config/environment';
+import IntlService from 'ember-intl/services/intl';
+import { tracked } from '@glimmer/tracking';
+import { action } from '@ember/object';
+import Store from '@ember-data/store';
+import { task } from 'ember-concurrency';
+
+// eslint-disable-next-line ember/use-ember-data-rfc-395-imports
+import DS from 'ember-data';
+
+import ProjectModel from 'irene/models/project';
+import DevicePreferenceModel from 'irene/models/device-preference';
+import ProjectAvailableDeviceModel from 'irene/models/project-available-device';
+
+export interface DevicePreferenceContext {
+  deviceTypes: DeviceType[];
+  selectedDeviceType?: DeviceType;
+  handleSelectDeviceType: (deviceType: DeviceType) => void;
+  selectedVersion: string;
+  devicePlatformVersions: string[];
+  handleSelectVersion: (version: string) => void;
+  isPreferredDeviceAvailable: boolean | null;
+}
+
+export interface ProjectPreferencesProviderSignature {
+  Args: {
+    project?: ProjectModel | null;
+    profileId?: number | string;
+    platform?: number;
+  };
+  Blocks: {
+    default: [DevicePreferenceContext];
+  };
+}
+
+type EnumObject = { key: string; value: number | string };
+type DeviceType = EnumObject;
+
+export default class ProjectPreferencesProviderComponent extends Component<ProjectPreferencesProviderSignature> {
+  @service declare intl: IntlService;
+  @service declare ajax: any;
+  @service('notifications') declare notify: NotificationService;
+  @service declare store: Store;
+
+  @tracked selectedVersion = '0';
+
+  @tracked selectedDeviceType?: DeviceType;
+
+  @tracked deviceTypes = ENUMS.DEVICE_TYPE.CHOICES;
+  @tracked devicePreference?: DevicePreferenceModel;
+
+  @tracked
+  devices: DS.AdapterPopulatedRecordArray<ProjectAvailableDeviceModel> | null =
+    null;
+
+  get tAnyVersion() {
```

Comment:
Position of getter should be after constructor.

Reply:
done